routes/drones.js

```diff
@@ -1,37 +1,49 @@
-const express = require('express');
-
-// require the Drone model here
+const express = require("express");
+const Drone = require("./../models/Drone.model");
 
 const router = express.Router();
 
-router.get('/drones', (req, res, next) => {
-  // Iteration #2: List the drones
-  // ... your code here
+router.get("/drones", (req, res, next) => {
+  Drone.find()
+    .then((allDronesFromDb) => {
+      res.render("drones/list", { allDronesFromDb });
+    })
+    .catch((err) => console.error(err));
 });
 
-router.get('/drones/create', (req, res, next) => {
-  // Iteration #3: Add a new drone
-  // ... your code here
+router.get("/drones/create", (req, res, next) => {
+  res.render("drones/create-form");
 });
 
-router.post('/drones/create', (req, res, next) => {
-  // Iteration #3: Add a new drone
-  // ... your code here
+router.post("/drones/create", (req, res, next) => {
+  const newDrone = req.body;
+  Drone.create(newDrone)
+    .then(() => {
+      res.redirect("/drones");
+    })
+    .catch((err) => console.error(err));
 });
 
-router.get('/drones/:id/edit', (req, res, next) => {
-  // Iteration #4: Update the drone
-  // ... your code here
+router.get("/drones/:id/edit", (req, res, next) => {
+  Drone.find({ _id: req.params.id })
+    .then((singleDrone) => res.render("drones/update-form", { singleDrone }))
+    .catch((err) => console.error(err));
 });
 
-router.post('/drones/:id/edit', (req, res, next) => {
-  // Iteration #4: Update the drone
-  // ... your code here
+router.post("/drones/:id/edit", (req, res, next) => {
+  const { name, propellers, maxSpeed } = req.body;
+  Drone.findByIdAndUpdate(
+    { _id: req.params.id },
+    { $set: { name: name, propellers: propellers, maxSpeed: maxSpeed } }
```

Comment:
You could have also used a shorthand notation `{ $set: { name, propellers, maxSpeed} }`